enumerable.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+    i = 0
+    while i < self.length
+      yield(self[i])
+      i += 1     
+    end
+    self
+  end
+  
+  def my_each_with_index
+    return enum_for unless block_given?
+    i =0
+    while i < self.length
+      yield(self[i], i)
+      i += 1   
+    end
+    self
+  end
+
+  def my_select
+    arr= []
+    x = 0
+    while x < self.length
+      arr << self[x] if yield(self[x]) == true
+      x += 1
+    end
+    arr
+  end
+
+  def my_all?
```

Comment:
false_block = proc { |num| num > 9 } }
Testing Range(5, 50).my_all?(&false_block) is not equal to Range(5,50).all?(&false_block)